pkg/services/pluginsintegration/angulardetectorsprovider/dynamic.go

```diff
@@ -184,6 +188,14 @@ func (d *Dynamic) IsDisabled() bool {
 	return !d.features.IsEnabled(featuremgmt.FlagPluginsDynamicAngularDetectionPatterns)
 }
 
+// randomSkew returns a random time.Duration between -ttl/2 and +ttl/2.
+// This can be added to backgroundJobInterval to skew it by a random amount.
```

Comment:
shouldn't this be always positive? Also, isn't the gap too big? If it's set to fetch this every hour and it can change up to 30min it could be confusing.

Reply:
Yes you're right it should always be positive, I have fixed it. I have also reduced the skew from 30m to 15m max